editor.planx.uk/src/@planx/components/Send/uniform/UniformPayload/model.ts

```diff
@@ -0,0 +1,302 @@
+            address?.county,
+          ].filter(Boolean) as string[],
+          "apd:Country": address?.country,
+          "apd:InternationalPostCode": address?.postcode,
+        },
+      };
+  };
+
+  private getAddressFromSiteAddress = (): ExternalAddress => ({
+    "common:InternationalAddress": {
+      "apd:IntAddressLine": this.siteAddress.title?.split(", "),
+      "apd:InternationalPostCode": this.siteAddress?.postcode,
+    },
+  });
+
+  private getNamespaces = () => ({
+    "_xmlns:portaloneapp":
+      "http://www.govtalk.gov.uk/planning/OneAppProposal-2006",
+    "_xmlns:xsi": "http://www.w3.org/2001/XMLSchema-instance",
+    "_xmlns:bs7666": "http://www.govtalk.gov.uk/people/bs7666",
+    "_xmlns:org": "http://www.govtalk.gov.uk/financial/OrganisationIdentifiers",
+    "_xmlns:pdt": "http://www.govtalk.gov.uk/people/PersonDescriptives",
+    "_xmlns:apd": "http://www.govtalk.gov.uk/people/AddressAndPersonalDetails",
+    "_xmlns:core": "http://www.govtalk.gov.uk/core",
+    "_xmlns:common": "http://www.govtalk.gov.uk/planning/OneAppCommon-2006",
+  });
+
+  private getRequiredFiles = (): FileAttachment[] => [
+    {
+      "common:Identifier": "N10049",
+      "common:FileName": "proposal.xml",
+      "common:Reference": "Schema XML File",
+    },
+    {
+      "common:FileName": "application.csv",
+      "common:Reference": "Other",
+    },
+  ];
+
+  private getUserUploadedFiles = (): FileAttachment[] =>
+    this.files.map((file) => {
+      const uniqueFilename = file.split("/").slice(-2).join("-");
+      return {
+        "common:FileName": uniqueFilename,
+        "common:Reference": "Other",
+      };
+    });
+
+  private setProposalCompletionDate = (): string => {
+    // ensure that date is valid and in yyyy-mm-dd format
+    let proposalCompletionDate =
+      this.passport.data?.["proposal.completion.date"];
+    if (proposalCompletionDate) {
+      proposalCompletionDate = new Date(proposalCompletionDate)
+        .toISOString()
+        .split("T")[0];
+    } else {
+      proposalCompletionDate = new Date(Date.now()).toISOString().split("T")[0];
+    }
+    return proposalCompletionDate;
+  };
+
+  private getPayment = (): Payment => {
+    const payment = this.passport.data?.[GOV_PAY_PASSPORT_KEY] as GovUKPayment;
+    return {
+      "common:PaymentMethod": "OnlineViaPortal",
+      "common:AmountDue": this.passport.data?.["application.fee.payable"] || 0,
+      "common:AmountPaid": payment?.amount || 0,
+      "common:Currency": "GBP",
+    };
+  };
+
+  public buildXML = (): string => {
+    const xmlDeclaration = {
+      _version: "1.0",
+      _encoding: "UTF-8",
+      _standalone: "yes",
+    };
+    const payload = {
+      "?xml": xmlDeclaration,
+      "portaloneapp:Proposal": this["portaloneapp:Proposal"],
+    };
+    const buildOptions: Partial<XmlBuilderOptions> = {
+      ignoreAttributes: false,
+      attributeNamePrefix: "_",
+      format: true,
+      suppressEmptyNode: true,
```

Comment:
is it right that `suppressEmptyNode: true` is what now handles still generating empty field placeholders for any "undefined" passport variables?

Reply:
Good question!

`suppressEmptyNode` renders nodes as `<something />` instead of `<something></something>` in the case of missing values.

This seemed like a sensible approach to help debugging and to more clearly indicate we don't have a value for something.